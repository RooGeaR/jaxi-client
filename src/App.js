import React from 'react';
import { StateProvider } from './hooks/state';
import { MutationForm, UpdateForm, List, Detail } from './pages/user/components';
import { 
	MutationForm as ProjectMutationForm, 
	UpdateForm as ProjectUpdateForm, 
	List as ProjectList, 
	Detail as ProjectDetail
} from './pages/project/components';
import { Panel } from './pages/panel/components';
import { MainLayout, NotFound } from './pages/common'
import { createUser, createProject } from './mutations';
import { Route, Switch } from 'react-router-dom';


const App = () => {
	
	const initialState = {
		callRefetch: false
	};
	  
	const reducer = (state, action) => {
		switch (action.type) {
		  case 'SET_CALL_REFETCH':
			return {
			  ...state,
			  callRefetch: action.callRefetch
			};
			
		  default:
			return state;
		}
	};

    return (
		<StateProvider initialState={initialState} reducer={reducer}>
		<div className="App">
			<Switch>
				<Route 
					exact 
					path='/'
					render={
						routeProps => (
							<Panel 
								{...routeProps}
								options={
									[
										{
											text: 'Users',
											icon: 'users',
											route: 'user'
										},
										{
											text: 'Projects',
											icon: 'folder-open',
											route: 'project'
										}
									]
								} 
							/>
						)
					} 
				/>

				<Route 
					exact 
					path='/user/create' 
					render={
						routeProps => (
							<MainLayout>
								<MutationForm {...routeProps} action={createUser} isNewRecord={true}/>
							</MainLayout>
						)
					} 
				/>

				<Route 
					exact 
					path='/user/update/:userId' 
					render={
						routeProps => (
							<MainLayout>
								<UpdateForm { ...routeProps } />
							</MainLayout>
						)
					} 
				/>

				<Route 
					exact 
					path='/user' 
					render={
						routeProps => (
							<MainLayout>
								<List { ...routeProps } />
							</MainLayout>
						)
					} 
				/>

				<Route 
					exact 
					path='/user/:userId' 
					render={
						routeProps => (
							<MainLayout>
								<Detail { ...routeProps } />
							</MainLayout>
						)
					} 
				/>

				<Route 
					exact 
					path='/project' 
					render={
						routeProps => (
							<MainLayout>
								<ProjectList { ...routeProps } />
							</MainLayout>
						)
					} 
				/>

				<Route 
					exact 
					path='/project/create' 
					render={
						routeProps => (
							<MainLayout>
								<ProjectMutationForm {...routeProps} action={createProject} isNewRecord={true}/>
							</MainLayout>
						)
					} 
				/>

				<Route 
					exact 
					path='/project/update/:projectId' 
					render={
						routeProps => (
							<MainLayout>
								<ProjectUpdateForm { ...routeProps } />
							</MainLayout>
						)
					} 
				/>

				<Route 
					exact 
					path='/project/:projectId' 
					render={
						routeProps => (
							<MainLayout>
								<ProjectDetail { ...routeProps } />
							</MainLayout>
						)
					} 
				/>

				<Route component={ NotFound } />
			</Switch>
		</div>
		</StateProvider>
    );
}

export default App;
