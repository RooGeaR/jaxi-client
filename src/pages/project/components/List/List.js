import React from 'react';
import { Link } from 'react-router-dom';
import { Query } from 'react-apollo';
import { projectListQuery } from '../../../../queries';
import DeleteButton from '../DeleteButton';
import { useGlobalState } from '../../../../hooks/state';
import styled from 'styled-components';

const Table = styled.table`
    background-color: #6699FF;

    th {
      color: white;
    }

    td {
      color: #3366CC;
    }
`;

const List = () => {

    const [ state, dispatch ] = useGlobalState();

    return (
        <Query query={ projectListQuery } >
          {({ data, loading, error, fetchMore, refetch }) => {
            if (loading) return <p>loading...</p>;
            if (error) return <p>{error.message}</p>
            
            if (state.callRefetch) {
              refetch();
              dispatch({
                  type: 'SET_CALL_REFETCH',
                  callRefetch: false
              });
            }
            
            return (
              <>
              <div className='row'>
                <div className='col-md-12'>
                <h2>Project list</h2>
                  <button type='button' className='btn btn-default pull-right'>
                    <Link to={`/project/create`}>
                      <span className='fa fa-plus'></span> Create
                    </Link>
                  </button>
                </div>  
              </div>

              <div className='row'>
                <div className={'col-md-12'}>

                  <Table className='table'>
                    <thead>
                      <tr>
                        <th>#</th>
                        <th>Title</th>
                        <th>Actions</th>
                      </tr>
                    </thead>
                    <tbody>
                      {
                        data.projects.map((project, i) => {
                          return (
                            <tr key={i}>
                              <td>{ i+1 }</td>
                              <td>{ project.title }</td>
                              <td>
                                <button>
                                  <Link to={`/project/${project.id}`}>
                                      <span className='fa fa-eye'></span>
                                  </Link>
                                </button>
                                <button>
                                  <Link to={`/project/update/${project.id}`}>
                                      <span className='fa fa-pencil'></span>
                                  </Link>
                                </button>
                                <DeleteButton 
                                  projectId={Number(project.id)} 
                                  refetch={refetch} 
                                />
                              </td>
                            </tr>
                          );
                        })
                      }
                    </tbody>
                  </Table>
                </div>
              </div>
              </>
            );
          }}
        </Query>
    );
}

export default List;