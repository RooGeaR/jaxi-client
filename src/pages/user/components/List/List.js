import React from 'react';
import { Link } from 'react-router-dom';
import { Query } from 'react-apollo';
import { userListQuery } from '../../../../queries';
import DeleteButton from '../DeleteButton';
import styled from 'styled-components';

const Table = styled.table`
    background-color: #6699FF;

    th {
      color: white;
    }

    td {
      color: #3366CC;
    }
`;

const List = () => {

    return (
        <Query query={ userListQuery } >
          {({ data, loading, error, fetchMore, refetch }) => {
            if (loading) return <p>loading...</p>;
            if (error) return <p>{error.message}</p>
            console.log('datalist',data);
            return (
              <>
              <div className='row'>
                <div className='col-md-12'>
                <h2>User list</h2>
                  <button type='button' className='btn btn-default pull-right'>
                    <Link to={`/user/create`}>
                      <span className='fa fa-plus'></span> Create
                    </Link>
                  </button>
                </div>  
              </div>

              <div className='row'>
                <div className={'col-md-12'}>

                  <Table className='table'>
                    <thead>
                      <tr>
                        <th>#</th>
                        <th>First name</th>
                        <th>Last name</th>
                        <th>Email</th>
                        <th>Username</th>
                        <th>Actions</th>
                      </tr>
                    </thead>
                    <tbody>
                      {
                        data.users.map((user, i) => {
                          return (
                            <tr key={i}>
                              <td>{ i+1 }</td>
                              <td>{ user.first_name }</td>
                              <td>{ user.last_name }</td>
                              <td>{ user.email }</td>
                              <td>{ user.username }</td>
                              <td>
                                <button>
                                  <Link to={`/user/${user.id}`}>
                                      <span className='fa fa-eye'></span>
                                  </Link>
                                </button>
                                <button>
                                  <Link to={`/user/update/${user.id}`}>
                                      <span className='fa fa-pencil'></span>
                                  </Link>
                                </button>
                                <DeleteButton 
                                  userId={Number(user.id)} 
                                  refetch={refetch} 
                                />
                              </td>
                            </tr>
                          );
                        })
                      }
                    </tbody>
                  </Table>
                </div>
              </div>
              </>
            );
          }}
        </Query>
    );
}

export default List;