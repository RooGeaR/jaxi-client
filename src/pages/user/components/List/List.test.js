import React from 'react';
import { shallow, render } from 'enzyme';
import List from './List';
import { StateProvider } from '../../../../hooks/state';
import { initialState, reducer } from '../../../../App';
import ApolloClient from 'apollo-boost';
import { ApolloProvider } from 'react-apollo';

const client = new ApolloClient({
    uri: "http://localhost:8088/graphql"
});

describe('List user', () => {
  it('should render correctly', () => {
    const component = shallow(
      <StateProvider initialState={initialState} reducer={reducer}>
        <List />
      </StateProvider>
    );
  
    expect(component).toMatchSnapshot();
  });
});

describe('List user', () => {
  it('should be render without error', () => {
    render(
      <ApolloProvider client={client}>
        <StateProvider initialState={initialState} reducer={reducer}>
          <List />
        </StateProvider>
      </ApolloProvider>
    );
    
  });
});
