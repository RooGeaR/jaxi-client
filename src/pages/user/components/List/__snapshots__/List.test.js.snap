// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List user should render correctly 1`] = `
<ContextProvider
  value={
    Array [
      Object {
        "callRefetch": false,
      },
      [Function],
    ]
  }
>
  <List />
</ContextProvider>
`;
